### PR TITLE
added the actual balance on overview page

### PR DESCRIPTION
**Pull request by vikas-cis-zz**
https://particl.atlassian.net/browse/PBD-226

### CODE REVIEW

**Comment by vikas-cis-zz** [comment]
Fixed #1440 

@AllienWorks  Probably you need to look the Pr, As possibly there need to be some css stuff if required , Below is the screenshot 

Thanks 
screen shot 2019-02-14 at 7 28 57 pm

**Comment by HenkSwardt** [comment]
@AllienWorks @vikas-cis Can we maybe change the wording of "ACTUAL BALANCE" to "SPENDABLE BALANCE" ?

**Comment by vikas-cis-zz** [comment]
@HenkSwardt yes ofcourse we can 

Thanks

**Comment by vikas-cis-zz** [comment]
Done !!! 

Thanks

**Comment by vikas-cis-zz** [comment]
https://particl.atlassian.net/browse/PBD-67

**Comment by vikas-cis-zz** [comment]
@AllienWorks Can you please have a look above Pr I think we need some css stuff to sort out the current change regarding spendable balance.

Thanks 
screen shot 2019-03-01 at 4 21 46 pm

**Comment by AllienWorks** [comment]
Looks good to me, no CSS changes necessary..

screenshot from 2019-03-04 11-14-13

**Comment by AllienWorks** [comment]
Styled & ready for merge.

screenshot from 2019-03-04 12-10-29

**Comment by AllienWorks** [comment]
OK, _now_ it is styled & working :D

screenshot from 2019-03-04 15-06-05

**Comment by vikas-cis-zz** [comment]
@AllienWorks Thank you I am gona check it out 

Thanks

**Comment by AllienWorks** [comment]
Layout fixed & styled. This PR absorbed changes from #1516.

Screenshot from 2019-03-20 15-32-33

Are we sure we don't have any code left-overs from the former solutions (the balance in top bar)? What I found in code I removed..

Otherwise ready for review @zaSmilingIdiot :)

File: src/app/wallet/wallet/balances/balance.component.ts [hunk 1]
```diff
@@ -15,10 +15,11 @@ import { takeWhile } from 'rxjs/operators';
 export class BalanceComponent implements OnInit, OnDestroy {
 
   @Input() type: string; // "total_balance", "anon_balance", "balance", "staked_balance", "blind_balance"
+  @Input() inSidebar: boolean = false; // located in sidebar?
 
   private log: any = Log.create(`balance.component ${this.type}`);
   private destroyed: boolean = false;
-
+  private availableBal: number = 0;
```
Comment (zaSmilingIdiot) [inline src/app/wallet/wallet/balances/balance.component.ts]: Any particular reason this is not of the 'Amount' type?

File: src/app/wallet/wallet/balances/balance.component.ts [hunk 2]
```diff
@@ -41,19 +43,38 @@ export class BalanceComponent implements OnInit, OnDestroy {
...
     }
 
     return this.type;
   }
 
+  listUnSpent(balance: number): void {
+    this._rpcState.observe('listunspent')
+      .pipe(takeWhile(() => !this.destroyed))
+      .subscribe(unspent => {
+          this.availableBal = 0;
```
Comment (zaSmilingIdiot) [inline src/app/wallet/wallet/balances/balance.component.ts]: Create a temporary variable of type number here (default assigned a value of 0), then set `this.availableBal` after the for-loop below, using the temp variable to create an Amount type.

File: src/app/wallet/wallet/send/send.component.html [hunk 3]
```diff
@@ -16,7 +16,7 @@
       </mat-tab>
     </mat-tab-group>
 
-    <div class="buttons">
+    <div class="buttons" *ngIf="send.input === 'anon' || type === 'sendPayment'">
```
Comment (zaSmilingIdiot) [inline src/app/wallet/wallet/send/send.component.html]: Prefer to use the `TxType` types if `send.input` is of the type `TxType`. 
For example: replace with something like: 
`send.input === TxType.ANON || type === 'sendPayment'`

File: src/app/wallet/wallet/send/send.component.ts [hunk 4]
```diff
@@ -18,27 +19,30 @@ import { TransactionBuilder, TxType } from './transaction-builder.model';
...
+  private destroyed: boolean = false;
+
   testnet: boolean = false;
   // UI logic
   @ViewChild('address') address: ElementRef;
   type: string = 'sendPayment';
   advanced: boolean = false;
   // TODO: Create proper Interface / type
   public send: TransactionBuilder;
+  private availableBal: number = 0;
```
Comment (zaSmilingIdiot) [inline src/app/wallet/wallet/send/send.component.ts]: Any particular reason this is not of the 'Amount' type?

File: src/app/wallet/wallet/send/send.component.ts [hunk 5]
```diff
@@ -68,13 +72,26 @@ export class SendComponent implements OnInit {
     this._rpcState.observe('getblockchaininfo', 'chain').pipe(take(1))
       .subscribe(chain => this.testnet = chain === 'test');
 
-    this.sendService.listUnSpent();
+    // Calculate Spendable balance
+    this._rpcState.observe('listunspent')
+      .pipe(takeWhile(() => !this.destroyed))
+      .subscribe(
+        unspent => {
+          this.availableBal = 0;
```
Comment (zaSmilingIdiot) [inline src/app/wallet/wallet/send/send.component.ts]: Create a temporary variable of type number here (default assigned a value of 0), then set this.availableBal after the for-loop below, using the temp variable to create an Amount type.

File: src/app/core/rpc/rpc-state/rpc-state.service.ts [hunk 6]
```diff
@@ -22,6 +22,7 @@ export class RpcStateService extends StateService implements OnDestroy {
     super();
 
     this.register('getwalletinfo', 5000);
+    this.register('listunspent', 1000, [0]);
```
Comment (zaSmilingIdiot) [inline src/app/core/rpc/rpc-state/rpc-state.service.ts]: This can in all likelihood be increased to 5000 ms (as per the getwalletinfo rpc)

**Review by zaSmilingIdiot — CHANGES_REQUESTED** [review]
The changes implemented in send.component.ts and balance.component.ts seem to be the same. Would it not be more applicable to move into a service and access the value from each component?

File: src/app/wallet/wallet/balances/balance.component.ts (hunk 1 above)
Comment (vikas-cis-zz) [reply src/app/wallet/wallet/balances/balance.component.ts]: availableBal is the temp to store total calculation of unspent amount, That's it

File: src/app/core/rpc/rpc-state/rpc-state.service.ts (hunk 6 above)
Comment (vikas-cis-zz) [reply src/app/core/rpc/rpc-state/rpc-state.service.ts]: ok

**Comment by vikas-cis-zz** [comment]
@AllienWorks I have added the spans and classes for the prices on listing page and preview page and now PR is all your's :) 

Thanks

**Comment by vikas-cis-zz** [comment]
https://particl.atlassian.net/browse/PBD-265 

The above changes also includes PBD-265

thanks

**Pull request merged**